### PR TITLE
[KYUUBI #497]upgrade `iceberg-spark3-runtime` to 0.11.1

### LINKED ISSUE #497: Bump up Apache Iceberg to 0.11.1
### 1. Describe the feature

See more detail here:

https://iceberg.apache.org/releases/#0110-release-notes
https://iceberg.apache.org/releases/#0111-release-notes

### 2. Motivation

Maybe wan can stop excluding iceberg tests for spark 3.1.1 with this artifact

### PR DESCRIPTION
**Pull request by chaosmin**
chaosmin [Closes #509](https://github.com/yaooqinn/kyuubi/pull/509)    [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### _Why are the changes needed?_

close #497 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -83,7 +83,7 @@
 
         <scalatest.version>3.0.3</scalatest.version>
         <iceberg.name>iceberg-spark3-runtime</iceberg.name>
-        <iceberg.version>0.11.0</iceberg.version>
+        <iceberg.version>0.11.1</iceberg.version>
```
Comment (yaooqinn) [inline pom.xml:86]: Would you mind if you can test this profile locally to see if the incompatibility of Iceberg has been gone ?https://github.com/yaooqinn/kyuubi/blob/master/.github/workflows/master.yml#L37 + remove -Dmaven.plugin.scalatest.exclude.tags=org.apache.kyuubi.tags.DataLakeTest

**Comment by yaooqinn** [comment]
Thanks for your first contribution and welcome! Just left one minor comment

File: pom.xml (hunk 1 above)
Comment (chaosmin) [reply pom.xml:86]: Iceberg not support Spark 3.1.1 yet, see https://github.com/apache/iceberg/issues/2311

Comment (yaooqinn) [reply pom.xml:86]: make sense

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master for v1.2.0

**Pull request closed without merging**